lib/index.js

```diff
@@ -0,0 +1,51 @@
+const path = require('path');
+const pathToRegexp = require('path-to-regexp');
+const url = require('url');
+
+const { loadDirectory } = require('./loader');
+
+/**
+ * iroute
+ * @param {Object} app Express instance
+ * @param {Object} options
+ * @param {Array} [options.interceptors=[]] Ordered interceptor list
+ * @param {String} [options.routesDir=$CWD/routes] Routes directory
+ * @param {Boolean} [options.throwError=false] Throw error of loading route
+ */
+module.exports = (app, options) => {
+  const { interceptors = [], routesDir = path.join(process.cwd(), 'routes'), throwError } = options;
+
+  const interceptorsMap = interceptors.reduce((result, interceptor) => {
+    result[interceptor.flag] = interceptor;
+    return result;
+  }, {});
+
+  interceptors.forEach((interceptor) => {
+    // each interceptor could use "exact=false" to use the broadly url-matching pattern of "app.use"
+    // it'd be "exact=true" by default to make everything compatible with previous version
+    const { exact = true, flag, path: interceptorPath } = interceptor;
```

Comment:
update the README for the new option `exact`?
https://github.com/springuper/express-iroute#interceptor-api

Reply:
and, could you add a test case for its non-default value?

Reply:
will do!